### ISSUE TITLE
add memtest on boot  

### ISSUE THREAD

**Issue by da1k1** [rescuezilla/rescuezilla#387]
Is it possible to add memtest on boot? I started using rescuezilla instead of other clone software some time ago and from time to time I'm missing ability to test RAM.

**Comment by Tarek-Hasan** [comment]
Please, update Memtest86+ to version 7.00.
Memtest86+ v6.00 which was released back in Oct, 2022, adds UEFI support and both DDR4 and DDR5 memory module support.

**Comment by Tarek-Hasan** [comment]
It would be nice to have memtest+ working on a system recovery OS like rescuzilla.

**Comment by shasheene** [comment]
Rescuezilla v2.5 is currently built on GitHub Actions servers to be released shortly with a build based on Ubuntu 24.04. I had assumed this would pull in memtest86+ v7, as that package is in Ubuntu 24.04 Noble: https://packages.ubuntu.com/noble/memtest86+.

But looking through the source code after receiving your comment ping 2 hours ago, the [memtest86+ package comes from the operating system package of the build _environment_](https://github.com/rescuezilla/rescuezilla/blob/master/src/build.sh#L322) (like other parts of the bootloader etc), and that environment remains an Ubuntu 22.04 (Jammy) based. So Rescuezilla v2.5.0 will be shipping the [memtest 5.31 from Jammy](https://packages.ubuntu.com/jammy/memtest86+).

I will upgrade the Docker container / build environment and pull in the memtest86+ package.

**Comment by Tarek-Hasan** [comment]
Thank you for your response.
Please, update the website with new version.

**Comment by shasheene** [comment]
Thanks for the reminder to update the website. Will do so after following up on some bugs that users have raised.

**Comment by Tarek-Hasan** [comment]
Hi,
Sorry to bother you, but v2.5.1 still ships with older version of memtest+. Also, it doesn't come with the latest Gparted v1.6.0 which was released back in February.

**Comment by shasheene** [comment]
> v2.5.1 still ships with older version of memtest+

Yep sorry, was an easy upgrade to pull in the latest version but I didn't get around to it. I'll do it now for the next Rescuezilla, which I'd expect will be done before the start of December, but I'll try and get something done sooner.

Answered your GParted question here: https://github.com/rescuezilla/rescuezilla/issues/136#issuecomment-2338529750
